### ISSUE TITLE
Add sigmoid, tanh and softmax functions to Tensor class

### ISSUE THREAD

**Issue by SIDD017** [SIDD017/PyTorch-Framework#4]
Ensure to make cuda kernels for the same
